### DISCUSSION TITLE
[Feature]: Different groups of week/month/year

### DISCUSSION THREAD

**Discussion by JVT038** [immich-app/immich#1661] [Feature Request]
### Feature detail

Currently, all items are grouped based on their day, but could it be possible to group items based on week / month / year?
Then I can instantly select all items of a specific month, instead of having to click through all the days of that month.
Something like this (I edited through inspect element):
![image](https://user-images.githubusercontent.com/47184046/189522809-e0a9b7e5-3384-4128-b7c5-f0a72503c00c.png)
So instead of per day, add an option to group them per week / month / year.

### Platform

Web & Mobile

**Comment by zackpollard** [comment]
Google Photos has a feature like this on Mobile, you can pinch to zoom on the gallery and it changes how specific the subheadings are, switching between zoomed day, day and month.

**Comment by vvzvlad** [comment]
I support, very much want such a feature. when every day one photo, it is very difficult to leaf through, instead of a bunch of photos on the line only one. In the mobile app it is perfectly realized

**Comment by photoserver-pro** [comment]
I believe, her you can find a similar request: https://github.com/immich-app/immich/discussions/12022

**Reply by mailinglists35** [reply]
I'll leave here link for arguments exposed in very large details,  since 12022 has been closed by admins in favor of current one

https://github.com/immich-app/immich/discussions/12022#discussioncomment-15439946

**Comment by photoserver-pro** [comment]
For large libraries, grouping results by Year / Month / Day isn’t a “nice to have” — it’s a basic navigation requirement. If a tag/keyword returns 6,000+ assets, or a person returns 15,000+, a flat infinite grid becomes practically unusable. The only way to browse those results efficiently is to chunk them by time, so you can jump straight to (for example) 2018 → May and work within a manageable subset.

This becomes even more important because Immich currently doesn’t offer a truly comprehensive faceted filter navigation across all views. Until that exists, time-based grouping is the most pragmatic way to make very large result sets explorable.

For archives at the scale of 500k+ assets, without time grouping, navigating big tag/person result lists is close to impossible in day-to-day use.

Would you consider implementing consistent Year/Month/Day grouping across result-heavy views (Tags, People, Search results, etc.) similar to the main Photos timeline?

![image](https://github.com/user-attachments/assets/cd4282b8-1f7e-42c8-b70d-d56f62fbef53)

**Comment by atahrijouti** [comment]
As this feature request is still open and the previous one (that's considered a duplicate was closed), I'll re-post my original message here

---

Photoprism, the closest product to Immich in terms of self hosting a photos catalogue for example has the following UI, which you can try it for yourself on this URL https://demo.photoprism.app/library/browse?view=cards&month=7&order=newest&public=true&quality=3

You click on the search settings in the search bar and you get a set of filters :
- You could filter by Year without specifying a month
- Then you can specify a month. 

<img width="1800" height="1034" alt="image" src="https://github.com/user-attachments/assets/1e3a1196-c7cd-4121-9140-f2e5108973db" />

In immich if I wanted to get July of 2024, I need to :
- Select the start date picker
- Select the year
- Select the month
- Select the day
- Select the end date picker
- Select the year
- Select the month
- Select the day

**Reply by alextran1502** [reply]
On the timeline, you can press “g” key then enter the month/year to go to that time frame
